static/js/types.ts

```diff
@@ -1,3 +1,17 @@
+export type MessageType = "private" | "stream";
+
+export type Message = {
+    match_content?: string;
+    match_subject?: string;
+    orig_subject?: string;
+    sender_email: string;
+    stream_id: number;
+    subject?: string;
+    to_user_ids?: string;
+    topic?: string;
+    type: MessageType;
```

Comment:
I suspect the question marks and associated non-null assertions can be avoided if we distinguish between the type of a message object before vs. after `process_new_message`.

Reply:
Yeah, something like `RawMessage` would be a good name for the pre-processing message object.